keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -3,703 +3,644 @@
-      csi: [3]
-        driver: cosi.sigs.k8s.io [4]
-        volumeAttributes: [5]
-          bucketAccessRequestName: <my-bar-name>
-```
-1. the service account may be needed depending on cloud IAM integration with kubernetes.
-1. the mount path is the directory where the app will read the credentials and endpoint.
-1. this is an inline CSI volume.
-1. the name of the cosi-node-adapter.
-1. information needed by the cosi-node-adapter to verify that the bucket has been provisioned.
+## Bucket Creation
 
-The contents of the secret generated by the provisioner is expected to have the following fields in the provider specific format:
+The following stakeholders are involved in the lifecycle of bucket creation:
 
-- Endpoint for the bucket. The Endpoint should encode the URL for the provider
-- Access Key: Only set when serviceAccount is empty 
-- Secret Key: Only set when serviceAccount is empty
+ - Users  - request buckets to be created for their workload
+ - Admins - setup COSI drivers in the cluster, and specify their configurations
 
-### Topology
+Here are the steps for creating a Bucket:
 
-![Architecture Diagram](images/arch.png)
+  ![COSI Bucket Creation](images/create-bucket.jpg)
 
-# Object Relationships
 
-The diagram below describes the relationships between various resources in the COSI ecosystem. 
+###### 1.  Admin creates BucketClass, User requests Bucket to be created
 
-![Object Relationships](images/object-rel.png)
+The BucketClass represents a set of common properties shared by multiple buckets. It is used to specify the driver for creating Buckets, and also for configuring driver-specific parameters. More information about BucketClass is [here](#bucketclass)
 
-# Workflows
-Here we describe the workflows used to automate provisioning of new and existing buckets, and the de-provisioning of these buckets.
+The BucketRequest is a request to create a Bucket. This resource is designed to specify properties that the application requires, such as the data protocol that this bucket should satisfy (i.e. s3, gcs or azure). More information about BucketRequest is [here](#bucketrequest)
 
-When the app pod is started, the kubelet will gRPC call _NodeStageVolume_ and _NodePublishVolume_, which are received by the cosi-node-adapter. The pod waits until the adapter responds to the gRPC requests. The adapter ensures that the target bucket has been provisioned and is ready to be accessed.
+```
+    BucketRequest - br1                                   BucketClass - bc1
+    |------------------------------|                      |----------------------------------|
+    | spec:                        |                      | spec:                            |
+    |   bucketClassName: bc1       |                      |   provisioner: s3.amazonaws.com  |
+    |   protocol: s3               |                      |   parameters:                    | 
+    |   deletionPolicy: delete     |                      |     key: value                   |
```

Comment:
deletion policy is not in the BR.

Reply:
will fix in the resource definition below